### PR TITLE
feat: support relative path with target config.

### LINKED ISSUE #1750: Support relativePath link set linkTarget
Hi
When I read the  docs of "Set target attribute for link" https://docsify.js.org/#/helpers?id=set-target-attribute-for-link 
I try a relative path to jump

```
[demo](/docs/front_end/summery_of_experience/data_use/assets/demo/index.html ':target=_blank')
```

but It be compiled as
```
<a href="/docs/front_end/summery_of_experience/data_use/assets/demo/index.html">demo</a>
```
It is not my expectation.

After then, I read the source code: "src\core\render\compiler\link.js" line 22
I discover that `docsifyjs` didnt support relative path linkTarget custom

so I make the issue in order to discuss the problem

thx

### PR DESCRIPTION
**Pull request by Koooooo-7**
## **Summary**

Support relative path set `target` config.
close #1750 

## **What kind of change does this PR introduce?**

Bugfix

## **For any code change,**

- [ ] Related documentation has been updated if needed
- [ ] Related tests have been updated or tests have been added

## **Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

## **Related issue, if any:**
#1750 

## **Tested in the following browsers:**

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

### CODE REVIEW

File: src/core/render/compiler/link.js [hunk 1]
```diff
@@ -28,6 +28,12 @@ export const linkCompiler = ({
       }
 
       href = router.toURL(href, null, router.getCurrentPath());
+
+      if (config.target) {
+        attrs.push(
+          href.indexOf('mailto:') === 0 ? '' : `target="${linkTarget}"`
```
Comment (jhildenbiddle) [inline src/core/render/compiler/link.js:34]: Does the empty string in the `attrs` array serve a purpose?

I'm not familiar with this part of the code so perhaps this is what needs to be done, but if the empty string does not serve a purpose (i.e., it's skipped when `attrs` is processed) then perhaps we should avoid it using a simple `if` statement:

```js
if (config.target) {
  if (href.indexOf('mailto:') !== 0) {
    attrs.push(`target="${linkTarget}"`)
  }
}
```

Or a single-line conditional like this:

```js
if (config.target) {
  href.indexOf('mailto:') !== 0 && attrs.push(`target="${linkTarget}"`);
}
```

**Review by jhildenbiddle — CHANGES_REQUESTED** [review]
One minor stylistic question, otherwise looks good. :)

File: src/core/render/compiler/link.js (hunk 1 above)
Comment (Koooooo-7) [reply src/core/render/compiler/link.js]: Got it !

**Review by jhildenbiddle — DISMISSED** [review]
LGTM!

File: src/core/render/compiler/link.js (hunk 1 above)
Comment (trusktr) [reply src/core/render/compiler/link.js]: Just my preference, but:

```js
if (config.target && href.indexOf('mailto:') !== 0) {
  attrs.push(`target="${linkTarget}"`);
}
```

**Comment by trusktr** [comment]
Why do we check if it doesn't start with `mailto:`? Does that break mailto links? 

I'm of the opinion that we shouldn't get in the way, should not change intent. So if someone writes a mailto link with a target, just like they can in regular HTML (`<a href="mailto:..." target="blank">`), we should just let it be.

It in other words, it shouldn't be our goal to change DOM conventions, but only to provide the interface to it as much as we can.

**Comment by jhildenbiddle** [comment]
@trusktr --

> Why do we check if it doesn't start with mailto:? Does that break mailto links?

I don't know the history, but my guess is that the original contributor was unaware of the usefulness of using the `target` attribute on `mailto:` links. The `target` attribute is ignored when the default email client is an external app (Apple Mail, Outlook, Thunderbird, etc) but it is honored when the default email client is web-based (Gmail, iCloud, Outlook.com, etc).

> I'm of the opinion that we shouldn't get in the way, should not change intent. So if someone writes a mailto link with a target, just like they can in regular HTML (`<a href="mailto:..." target="blank">`), we should just let it be.

Agreed, but this is a separate issue. My vote would be to review this PR for its intended purpose and create another PR to "fix" the target attribute on `mailto:` links.

**Review by jhildenbiddle — APPROVED** [review]

**Comment by Koooooo-7** [comment]
@trusktr  Hi, as @jhildenbiddle  said, it is the original contributor did, I wanna keep it incase of any unpredicted changes.
If we could change the `mailto:` issue, I agree that we can have a new one  to fix that .

**Comment by jhildenbiddle** [comment]
@Koooooo-7 --

Since this has been targeted for v5, can we update this branch and get one more review/approval? @sy-records? @trusktr?

**Comment by jhildenbiddle** [comment]
👆 @sy-records

**Review by sy-records — APPROVED** [review]

**Pull request merged**